pkg/openwsn/doc.txt

```diff
@@ -0,0 +1,413 @@
+ how to route, instead it is `OpenVisualizer` which generates an SRH (Source
+ Routing Header ), attaches to the incoming packet and sends it down the tree.
+
+ ## Hardware abstraction implementation
+
+ Following, details about the implementation of selected hardware modules.
+
+ ### sctimer
+
+ The `sctimer` ("single compare timer") in OpenWSN is the lowest timer
+ abstraction which is used by the higher layer timer module `opentimers`. In
+ the end it is responsible for scheduling on the MAC layer. To enable low power
+ energy modes, this timer usually uses the RTC (real time clock) or RTT (real
+ time timer) module.
+
+ This port has two possible implementations or sctimer, one on top of
+ periph_rtt and the another on top of ztimer, `sctimer_rtt` and
+ `sctimer_ztimer` respectively. If possible `ztimer` should be preferred,
+ being a virtual timer it will allow RIOT applications/modules to use the
+ low level RTT timer. But ztimer (any virtual timer) has some overhead
+ which can be costly depending on the different platforms used, specially
+ when `openserial` is required. More on this in openserial and Known Issues.
+
+ #### sctimer_ztimer
+
+ In order to get the most portable code, this implementation uses
+ ztimer and defines a new `ztimer_clock` (`ZTIMER_32768`) that operates
+ at 32768Khz to have a resolution of ~30usec/tick (same as OpenWSN).
+
+ When available `ZTIMER_32768` will be built on top of `periph_rtt` to
+ get low power capabilities. If not it will be built on top of a
+ regular timer. In either case it will be shifted up if the base
+ frequency is lower than 32768Hz or frac if higher.
+
+ When next interrupt to schedule is already late, current time,
+ Implementations in OpenWSN directly trigger a hardware interrupt.
+ Until able to trigger sw isr directly a callback is set 0 ticks in
+ the future, which internally will be set to `now + RTT_MIN_OFFSET`.
+
+ #### sctimer_rtt
+
+ In order to reduce overhead this implementation uses bare RTT. It
+ expects a RTT running at 32768Hz to have a resolution of ~30usec/tick
+ (same as OpenWSN). If `RTT_FREQUENCY` is lower than 32768Hz then a
+ simple time-division mechanism will be used to speed up the clock.
+ This only works if `RTT_FREQUENCY` is 32768Hz/2.
+
+ When next interrupt to schedule is already late, current time,
+ Implementations in OpenWSN directly trigger a hardware interrupt.
+ Until able to trigger sw isr directly a callback is set
+ `RTT_MIN_OFFSET` ticks in the future.
+
+ ### radio
+
+ The radio adaptation runs in its own thread with the highest priority
+ (`THREAD_PRIORITY_MAIN - 4`) and maps to RIOT's `netdev` API.
+
+ Hardware MAC layer features such as CSMA/CA, ACK handling and retransmissions
+ are handled by OpenWSN, so the radio driver must support disabling AUTOACK
+ and CSMA handling by the hardware. Frame filtering must as well be disabled.
+
+ The radio adaptation preloads the buffer so `NETOPT_PRELOADING` must be
```

Comment:
```suggestion
 The radio adaption preloads the buffer so `NETOPT_PRELOADING` must be
```

Reply:
Hmm adaption is rarely used in English currently, I would prefer sticking to adaptation, same below.

Reply:
I think both words are valid.

And also adaption seems to be more widely used in RIOT:
```
git grep adaption | wc -l
119
git grep adaptation | wc -l
14

```

Reply:
Sorry won't change.